### PR TITLE
Make the changelog update script more robust

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The `split-changelog` script did not recognize a cherry pick release, so manual changes had to me made for the `Release-2023-08-25` PR.

# Changes
- Loosen the regex that detects releases in the changelog, so that comments after the proposal number do not matter.

# Tests

Head of main gives this spurious diff:
```
ax@sinkpad:~/dfn/nns-dapp/branches/test-changelog-script (53:53)$ scripts/nns-dapp/split-changelog
patching file /home/max/dfn/nns-dapp/branches/test-changelog-script/CHANGELOG-Nns-Dapp.md
max@sinkpad:~/dfn/nns-dapp/branches/test-changelog-script (53:59)$ git diff Release-2023-08-25 CHANGELOG-Nns-Dapp.md
diff --git a/CHANGELOG-Nns-Dapp.md b/CHANGELOG-Nns-Dapp.md
index 180f6247d..c8cb4460a 100644
--- a/CHANGELOG-Nns-Dapp.md
+++ b/CHANGELOG-Nns-Dapp.md
@@ -40,6 +40,12 @@ The NNS Dapp is released through proposals in the Network Nervous System. Theref
 
 #### Security
 
+## Proposal 124252 (cherry-pick)
+
+### Application
+
+#### Fixed
+
 ## Proposal 124280
 
 ### Application
```

With the change in this PR we get:
```
max@sinkpad:~/dfn/nns-dapp/branches/test-changelog-script (54:31)$ git diff Release-2023-08-25 CHANGELOG-Nns-Dapp.md
diff --git a/CHANGELOG-Nns-Dapp.md b/CHANGELOG-Nns-Dapp.md
index 180f6247d..608c5ed88 100644
--- a/CHANGELOG-Nns-Dapp.md
+++ b/CHANGELOG-Nns-Dapp.md
@@ -113,6 +113,7 @@ The NNS Dapp is released through proposals in the Network Nervous System. Theref
 
 * Fixed some tests that depended on execution order.
 * [CVE-2023-38497](https://blog.rust-lang.org/2023/08/03/cve-2023-38497.html): Update Rust from version `1.71.0` to `1.71.1`.
+
 ## Proposal 124252 (cherry-pick)
 
 ### Application
max@sinkpad:~/dfn/nns-dapp/branches/test-changelog-script (54:38)$ git status
```
Note: There was a curious blank line change.  Compare with the line above.  This PR appears to undo that strange change.

# Todos

- [ ] Add entry to changelog (if necessary).

### CODE REVIEW

File: scripts/nns-dapp/split-changelog [hunk 1]
```diff
@@ -61,7 +61,7 @@ fi
 # Add 1 to start after the title.
 START_OF_UNRELEASED_SECTION="$(($(grep -n -m 1 '^## Unreleased$' "$OLD_CHANGELOG" | cut -d: -f1) + 1))"
 # Subtract 1 to end before the next title.
-END_OF_UNRELEASED_SECTION="$(($(grep -n -m 1 '^## Proposal [0-9]\+$' "$OLD_CHANGELOG" | cut -d: -f1) - 1))"
+END_OF_UNRELEASED_SECTION="$(($(grep -n -m 1 '^## Proposal [0-9]\+' "$OLD_CHANGELOG" | cut -d: -f1) - 1))"
```
Comment (bitdivine) [inline scripts/nns-dapp/split-changelog:64]: Note: The + could be deleted if wanted.

Comment (dskloetd) [reply scripts/nns-dapp/split-changelog:64]: Either way is fine.

**Review by dskloetd — APPROVED** [review]
Thanks!

**Comment by dskloetd** [comment]
Maybe add a change log entry?

**Pull request merged**